lib/create-http-client.js

```diff
@@ -17,38 +23,73 @@ import rateLimit from './rate-limit'
  * @prop {Object=} headers - Additional headers
  * @return {Object} Initialized axios instance
  */
-export default function createHttpClient (axios, httpClientParams) {
-  const {space, accessToken, insecure, host, defaultHostname, httpAgent, httpsAgent, proxy, retryOnError, timeout} = httpClientParams
-  let {headers} = httpClientParams
-  let [hostname, port] = (host && host.split(':')) || []
-  hostname = hostname || defaultHostname
-  port = port || (insecure ? 80 : 443)
-  let baseURL = `${insecure ? 'http' : 'https'}://${hostname}:${port}/spaces/`
-  if (space) {
-    baseURL += space + '/'
+export default function createHttpClient (axios, options) {
+  const defaultConfig = {
+    insecure: false,
+    retryOnError: true,
+    logHandler: (level, data) => {
+      if (level === 'error') {
+        const title = [data.name, data.message].filter((a) => a).join(' - ')
+        console.error(`[error] ${title}`)
+        console.error(data)
+        return
+      }
+      console.log(`[${level}] ${data}`)
+    },
+    // Passed to axios
+    headers: {},
+    httpAgent: false,
+    httpsAgent: false,
+    timeout: 30000,
+    proxy: false
+  }
+  const config = Object.assign({}, defaultConfig, options)
+
+  if (!config.accessToken) {
+    throw new TypeError(`'accessToken' option is missing`)
```

Comment:
let's use the same error as in [here ](https://github.com/contentful/contentful-management.js/blob/master/lib/contentful-management.js#L54)